paddle/framework/error.h

```diff
@@ -122,15 +117,23 @@ class Error {
    */
   bool isOK() const { return msg_ == nullptr; }
 
-  /**
-   * @brief check this status by glog.
-   * @note It is a temp method used during cleaning Paddle code. It will be
-   *       removed later.
-   */
-  void check() const { CHECK(this->isOK()) << msg(); }
+  bool operator==(const Error &o) const {
```

Comment:
@typhoonzero By using `msg_` a shared_ptr, the `operator==` will be fast for predefined Error.